src/activity_item_handling.cpp

```diff
@@ -488,11 +488,9 @@ static void move_items( const tripoint &src, bool from_vehicle,
         indices.pop_back();
         quantities.pop_back();
 
-        item *temp_item = nullptr;
-
-        temp_item = ( from_vehicle == true ) ?
-                    g->m.item_from( s_veh, s_cargo, index ) :
-                    g->m.item_from( source, index );
+        item *temp_item = ( from_vehicle == true ) ?
```

Comment:
Btw, can we just use `( from_vehicle )` instead of `( from_vehicle == true )`?

Reply:
There are several more of similar things:

````
  \src\advanced_inv.cpp (1 hit)
	Line 1319:             int amount = (item_it->count_by_charges() == true) ? item_it->charges : 1;
Search "== true ) ? " (2 hits in 1 file)
  \src\advanced_inv.cpp (2 hits)
	Line 580:         return ( can_store_in_vehicle() == true ) ? veh->get_items( vstor ).size() : 0;
	Line 1092:     draw_border( w, ( active == true ) ? BORDER_COLOR : c_dark_gray );
````

Reply:
> Btw, can we just use ( from_vehicle ) instead of ( from_vehicle == true )?

Yes, please, and the parentheses are not needed either: `foo = bar ? a : b`. and you could even make `temp_item` const.

Reply:
>and you could even make temp_item const.

I guess it is not allowed - `temp_item->charges` is being changed later in the code:

````
temp_item->charges = quantity;
````